pkg/linter/linter_test.go

```diff
@@ -138,6 +139,69 @@ or := 1
 			filename:      "p.rego",
 			expViolations: []string{"top-level-iteration"},
 		},
+		{
+			name: "ignore all",
+			userConfig: &config.Config{
+				Defaults: config.Defaults{
+					Global: config.Default{
+						Level: "ignore",
+					},
+				},
+			},
+			filename:      "p.rego",
+			expViolations: []string{},
+		},
+		{
+			name: "ignore all but bugs",
+			userConfig: &config.Config{
+				Defaults: config.Defaults{
+					Global: config.Default{
+						Level: "ignore",
+					},
+					Categories: map[string]config.Default{
+						"bugs": {Level: "error"},
+					},
+				},
+				Rules: map[string]config.Category{
+					"bugs": {"rule-shadows-builtin": config.Rule{Level: "ignore"}},
+				},
+			},
+			filename:      "p.rego",
+			expViolations: []string{"top-level-iteration"},
+		},
+		{
+			name: "ignore style, no global default",
+			userConfig: &config.Config{
+				Defaults: config.Defaults{
+					Categories: map[string]config.Default{
+						"bugs":  {Level: "error"},
+						"style": {Level: "ignore"},
+					},
+				},
+				Rules: map[string]config.Category{
+					"bugs": {"rule-shadows-builtin": config.Rule{Level: "ignore"}},
+				},
+			},
+			filename:      "p.rego",
+			expViolations: []string{"top-level-iteration"},
+		},
+		{
+			name: "set level to warning",
+			userConfig: &config.Config{
+				Defaults: config.Defaults{
+					Global: config.Default{
+						Level: "warning", // will apply to all but style
+					},
+					Categories: map[string]config.Default{
+						"style": {Level: "error"},
+					},
+				},
+				Rules: map[string]config.Category{},
+			},
+			filename:      "p.rego",
+			expViolations: []string{"opa-fmt", "top-level-iteration", "rule-shadows-builtin"},
+			expLevels:     []string{"error", "warning", "warning"},
+		},
```

Comment:
Test cases are good! I wonder if we should add something (e2e) for the case of "category disabled via config file, enabled via CLI" or similar... wouldn't need to be extensive but just something to ensure CLI args have precedence.

Reply:
Yeah I can look into doing that since we don't have that covered elsewhere